src/Pages/FriendsButton.js

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios'
+
+
+export default function FriendsButton({ otherUserId }) {
+
+    const [buttonText, setButtonText] = useState("Send friend request")
+    const [errors, setError] = useState(false)
+    // const [id, setId] = useState('');
+    // const [otherId, setOtherId] = useState('');
+
+    //props should be replaced by otherUserId
+    useEffect(() => {
+        console.log("mounted ")
+        axios
+            .get(`/getFriendsStatus/${otherUserId}`)
+
+
+            //check in the backend the current status of the friend request.
+            // Is there an existing friend request between a given pair of users ?
+
+            //     If there is a request, has it been accepted ?
+
+            //         If there is a request and it has not been accepted, who is the sender and who is the receiver ?
+            //     A simple way to accomplish this would be to create a table for friend requests that has columns for the id of the sender, the id of the recipient, and a boolean indicating whether or not the request has been accepted.When one user sends another a friend request, a row would be inserted with the ids of the sender and receiver in the appropriate columns and the boolean set to false.When a user accepts a friend request, the appropriate row would be updated to set the boolean to true.When a user unfriends or cancels a pending request, the row for the request can be deleted(deleting these rows means that we will lose potentially valuable historical information, but that is probably acceptable for our purposes).
+
+
+            .then(data => {
+                //define the orignal "state" of the message 
+                if (data && data.addFriend === true) {
+                    setButtonText("Send friend request")
+                } else if (data && data.unfriend === true) {
+                    setButtonText("Unfriend")
+                } else if (data && data.acceptFriend === true) {
+                    setButtonText("Accept Friend Request")
+                } else if (data && data.cancelFrequest === true) {
+                    setButtonText("Cancel Friend Request")
+                } else {
+                    setError(true)
+                }
+            })
+    }, [])
+
```

Comment:
I was actually thinking about this functionality as well.  Something to discuss when we talk about the backend stuff.  Will be a good opportunity to chat with Olivia.  I think getting user data stored and being able to add/remove friends and search the user database is probably our first goal?

Reply:
I don't really mind which one we decide to do first. I could actually write a draft on how to proceed in the db. I checked some MongoDB videos today to get familiar with it.